libraries/ms-common/inc/ms_test_helpers.h

```diff
@@ -55,16 +55,16 @@
 #define MS_TEST_HELPER_ASSERT_EVENT(event, e_id, e_data) \
   ({                                                     \
     TEST_ASSERT_OK(event_process(&(event)));             \
-    TEST_ASSERT_EQUAL((e_id), event.id);                 \
-    TEST_ASSERT_EQUAL((e_data), event.data);             \
+    TEST_ASSERT_EQUAL((e_id), (event).id);               \
```

Comment:
How did this fix your test? What difference does the bracket make? I'm interested to know what happens when there is a bracket and when there is no bracket.

Reply:
Macros work by simple substitution, so whatever you put in the place of the `event` parameter gets substituted whenever the C preprocessor sees `event`. With a single identifier this works fine with no brackets, but I was passing `*e`, which caused replacements of e.g. `*e.id`, which tries to dereference `e.id` and fails. Putting brackets fixes the precedence issue. It's generally good practice to put brackets around every macro parameter (and the macro itself) every time you use them.